HW17/package1/Refueling.java

```diff
@@ -0,0 +1,30 @@
+package package1;
+
+import java.util.concurrent.Semaphore;
+
+public class Refueling implements Runnable{
+
+    final PetrolStation station;
+    Semaphore semaphore;
+    String carName;
+
+    public Refueling(PetrolStation station, Semaphore semaphore, String carName) {
+        this.station = station;
+        this.semaphore = semaphore;
+        this.carName = carName;
+    }
+
+    @Override
+    public void run() {
+        try {
+            semaphore.acquire();
+            System.out.printf("%s at the station\n", carName);
+            synchronized (station){
+                station.doRefuel(carName);
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        semaphore.release();
```

Comment:
if any Exception occurs in the try block (after `semaphore.acquire()`) then this resource will never be released. Best practice keep it in final block.